### PR TITLE
Create Spring Boot-based archetype (v14)

### PR DESCRIPTION
**Pull request by tarekoraby**
Address #176 for v14

### CODE REVIEW

**Review by manolo — CHANGES_REQUESTED** [review]
- You need to add `<module>vaadin-archetype-spring-application</module>` to the root `pom.xml`
- I created a project with the generated archetype and I get compilation errors
```
Define value for property 'groupId': a.b.c
Define value for property 'artifactId': test
Define value for property 'version' 1.0-SNAPSHOT: :
Define value for property 'package' a.b.c: :
Confirm properties configuration:
groupId: a.b.c
artifactId: test
version: 1.0-SNAPSHOT
package: a.b.c
 Y: : Y
```
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project vaadin-archetype-spring-application: Compilation failure: Compilation failure:
[ERROR] /private/tmp/kk/ar/test/src/main/java/a/b/c/views/MainLayout.java:[9,40] package com.vaadin.flow.component.avatar does not exist
[ERROR] /private/tmp/kk/ar/test/src/main/java/a/b/c/views/MainLayout.java:[31,22] cannot find symbol
[ERROR]   symbol:   method themeFolder()
[ERROR]   location: @interface com.vaadin.flow.theme.Theme
[ERROR] /private/tmp/kk/ar/test/src/main/java/a/b/c/views/MainLayout.java:[31,1] annotation @com.vaadin.flow.theme.Theme is missing a default value for the element 'value'
[ERROR] /private/tmp/kk/ar/test/src/main/java/a/b/c/views/MainLayout.java:[54,9] cannot find symbol
[ERROR]   symbol:   class Avatar
[ERROR]   location: class a.b.c.views.MainLayout
[ERROR] /private/tmp/kk/ar/test/src/main/java/a/b/c/views/MainLayout.java:[54,29] cannot find symbol
[ERROR]   symbol:   class Avatar
[ERROR]   location: class a.b.c.views.MainLayout
[ERROR] -> [Help 1]
```

**Comment by manolo** [comment]
The generated pom should have the following changes, since it's not setting correctly: project version, groupId, artifactId,  name, and the property vaadin.version

```
--- a/pom.xml
+++ b/pom.xml
@@ -1,15 +1,15 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
-    <groupId>com.vaadin</groupId>
-    <artifactId>vaadin-archetype-spring-application</artifactId>
-    <name>Project base for Spring Boot and Vaadin Flow</name>
-    <version>14.0-SNAPSHOT</version>
+    <groupId>a.b.c</groupId>
+    <artifactId>foo</artifactId>
+    <name>My Foo App</name>
+    <version>1.0-SNAPSHOT</version>
     <packaging>jar</packaging>
 
     <properties>
         <java.version>8</java.version>
-        <vaadin.version>14.0.0</vaadin.version>
+        <vaadin.version>14.4.6</vaadin.version>
     </properties>
 
     <parent>
@@ -55,7 +55,7 @@
             <dependency>
                 <groupId>com.vaadin</groupId>
                 <artifactId>vaadin-bom</artifactId>
-                <version>14.0.0</version>
+                <version>${vaadin.version}</version>
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
@@ -174,7 +174,7 @@
             <plugin>
                 <groupId>com.vaadin</groupId>
                 <artifactId>vaadin-maven-plugin</artifactId>
-                <version>14.0.0</version>
+                <version>${vaadin.version}</version>
                 <executions>
                     <execution>
                         <goals>
@@ -195,7 +195,7 @@
                     <plugin>
                         <groupId>com.vaadin</groupId>
                         <artifactId>vaadin-maven-plugin</artifactId>
-                        <version>14.0.0</version>
+                        <version>${vaadin.version}</version>
                         <executions>
                             <execution>
                                 <goals>
```

File: vaadin-archetype-spring-application/src/main/resources/archetype-resources/src/main/java/views/MainLayout.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import com.vaadin.flow.router.PageTitle;
+import ${package}.views.MainLayout;
+import ${package}.views.helloworld.HelloWorldView;
+import ${package}.views.about.AboutView;
+
+/**
+ * The main view is a top-level placeholder for other views.
+ */
+@PWA(name = "My App", shortName = "My App", enableInstallPrompt = false)
+@Theme(themeFolder = "${artifactId}")
```
Comment (manolo) [inline vaadin-archetype-spring-application/src/main/resources/archetype-resources/src/main/java/views/MainLayout.java:34]: This needs a value in 14.x

But anyways, the other archetype produces an app with just a simple view, why this creates a full featured app, I think it should be just an skeleton like the other, for complicated apps we have the start.vaadin.com

**Review by manolo — CHANGES_REQUESTED** [review]

File: vaadin-archetype-spring-application/src/main/resources/archetype-resources/src/main/java/views/MainLayout.java (hunk 1 above)
Comment (tarekoraby) [reply vaadin-archetype-spring-application/src/main/resources/archetype-resources/src/main/java/views/MainLayout.java:34]: Fixed the themeFolder issue.
I wouldn't call this a full-featured app. I regard this as a skeleton with the basic building blocs of a Vaadin app. But in any case, we should aim to generate a consistent skeleton for Spring and plain Java projects. I'll change this one accordingly.

**Review by manolo — DISMISSED** [review]

**Review by manolo — APPROVED** [review]

**Pull request merged**